contracts/StorageAccessible.sol

```diff
@@ -79,23 +106,21 @@ contract StorageAccessible {
     function simulateDelegatecallInternal(
         address targetContract,
         bytes memory calldataPayload
-    ) external returns (bytes memory response) {
-        bool success;
-        (success, response) = targetContract.delegatecall(
-            calldataPayload
-        );
-        revertWith(abi.encodePacked(response, success));
-    }
-
-    function revertWith(bytes memory response) internal pure {
+    ) external {
         assembly {
-            revert(add(response, 0x20), mload(response))
-        }
-    }
+            let success := delegatecall(
+                gas(),
+                targetContract,
+                add(calldataPayload, 0x20),
+                mload(calldataPayload),
+                0,
+                0
+            )
 
-    function setLength(bytes memory buffer, uint256 length) internal pure {
-        assembly {
-            mstore(buffer, length)
+            mstore(0x00, success)
```

Comment:
Maybe we should document this layout `success:bool || data_length:uint256 || data:bytes`